code/modules/mob/living/carbon/human/init_signals.dm

```diff
@@ -38,3 +41,25 @@
 	if(name != voice_name)
 		voice_name += " (as [get_id_name("Unknown")])"
 	stored_name[NAME_PART_INDEX] = voice_name
+
+/mob/living/carbon/human/proc/on_fat(datum/source)
+	SIGNAL_HANDLER
+	hud_used?.hunger?.update_appearance()
+	mob_mood?.update_nutrition_moodlets()
+
+	if(HAS_TRAIT(src, TRAIT_FAT))
+		add_movespeed_modifier(/datum/movespeed_modifier/obesity)
+	else
+		remove_movespeed_modifier(/datum/movespeed_modifier/obesity)
+
+/mob/living/carbon/human/proc/on_nohunger(datum/source)
+	SIGNAL_HANDLER
+	// When gaining NOHUNGER, we restore nutrition to normal levels, since we no longer interact with the hunger system
+	if(HAS_TRAIT(src, TRAIT_NOHUNGER))
+		set_nutrition(NUTRITION_LEVEL_FED, forced = TRUE)
+		satiety = 0
+		overeatduration = 0
+		REMOVE_TRAIT(src, TRAIT_FAT, OBESITY)
+	else
+		hud_used?.hunger?.update_appearance()
```

Comment:
early return instead of if/else here imo

Reply:
this is misleading, i accidentally omitted another line in the `else` condition but i still think an early return would be far more clear

Reply:
I don't think inverting the conditions would be much clearer